### PR TITLE
Use Lettuce for Orbit Messaging

### PR DESCRIPTION
**Pull request by tmccnnll**
Change Orbit messaging to use [Lettuce](https://github.com/lettuce-io/lettuce-core) instead of Redisson.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -39,11 +39,11 @@ THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
     <groupId>cloud.orbit</groupId>
     <artifactId>orbit-redis-cluster</artifactId>
-    <version>1.2.8-SNAPSHOT</version>
+    <version>1.2.9-SNAPSHOT</version>
```
Comment (ceckhardt) [inline pom.xml]: So this should likely be 1.2.8.4-SNAPSHOT since this was forked from 1.2.8 and we already have a 1.2.8.3

Comment (tmccnnll) [reply pom.xml]: Makes sense. I'm going to see what Joe says about the preferred version number in the PR I have open in the base project before making any changes.

File: src/main/java/cloud/orbit/actors/cluster/impl/lettuce/LettuceOrbitClient.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+        if (this.redisSubscribingConnection.isOpen())
+        {
+            this.redisSubscribingConnection.addListener(messageListener);
+            final RedisFuture<Void> subscribeResult = this.redisSubscribingAsyncCommands.subscribe(channelId);
+            return (CompletableFuture)subscribeResult;
+        }
+        else
+        {
+            logger.error("Error subscribing to channel [{}]", channelId);
+            return new CompletableFuture<>();
```
Comment (jpennell) [inline src/main/java/cloud/orbit/actors/cluster/impl/lettuce/LettuceOrbitClient.java]: I believe this `CompletableFuture` will never complete. 
Might be better to return:
```
CompletableFuture.completedFuture(null)
```
or 
```
final CompletableFuture<Void> result = new CompletableFuture<>();
result.completeExceptionally(new IllegalStateException("Error subscribing to channel...."));
return result;
```
?

**Review by jpennell — APPROVED** [review]

File: src/main/java/cloud/orbit/actors/cluster/impl/lettuce/FstSerializedObjectCodec.java [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+        try
+        {
+            byte[] array = new byte[bytes.remaining()];
+            bytes.get(array);
+            FSTObjectInput in = fstConfig.get().getObjectInput(new ByteArrayInputStream(array));
+            return in.readObject();
+        }
+        catch (Exception e)
+        {
+            return null;
```
Comment (JordanAngold) [inline src/main/java/cloud/orbit/actors/cluster/impl/lettuce/FstSerializedObjectCodec.java]: Is this exception-discarding behaviour analogous to the existing Redisson implementation?

Comment (tmccnnll) [reply src/main/java/cloud/orbit/actors/cluster/impl/lettuce/FstSerializedObjectCodec.java]: Throwing an `IllegalStateException` seems to be the preferred approach. I'm basing this on what I see  in Lettuce's [`CompressionCodec`](https://github.com/lettuce-io/lettuce-core/blob/master/src/main/java/com/lambdaworks/redis/codec/CompressionCodec.java) and [`StringCodec`](https://github.com/lettuce-io/lettuce-core/blob/master/src/main/java/com/lambdaworks/redis/codec/StringCodec.java)

File: src/main/java/cloud/orbit/actors/cluster/impl/lettuce/LettuceOrbitClient.java (hunk 2 above)
Comment (tmccnnll) [reply src/main/java/cloud/orbit/actors/cluster/impl/lettuce/LettuceOrbitClient.java]: I like the latter approach.

**Pull request merged**